cmd/server/main.go

```diff
@@ -83,11 +90,27 @@ func main() {
 	}
 
 	go func() {
-		fmt.Println(s.ListenAndServe())
+		if err := httpServer.ListenAndServe(); err != http.ErrServerClosed {
+			log.Err(err).Msg("http server error received")
+			stop()
+		}
 	}()
 
-	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
-	defer stop()
+	grpcServer := grpc.NewServer()
```

Comment:
the GRPC server, as well as HTTP, accepts options - https://pkg.go.dev/google.golang.org/grpc#ServerOption
I wouldn't recommend any particular now, but it is worth looking at and remembering that it is there.

Reply:
also useful could be to have recovery middleware you could find such in this repo https://github.com/grpc-ecosystem/go-grpc-middleware